src/test/java/com/puppycrawl/tools/checkstyle/internal/ArchUnitTest.java

```diff
@@ -74,4 +95,35 @@ public void nonProtectedCheckMethodsTest() {
 
         checkMethodsShouldNotBeProtectedRule.check(importedClasses);
     }
+
+    /**
+     * The goal is to ensure all classes in api package are not dependent on classes in util
+     * pacakages.
+     *
+     * @noinspection JUnitTestMethodWithNoAssertions
+     * @noinspectionreason JUnitTestMethodWithNoAssertions - asserts in callstack,
+     *     but not in this method
+     */
+    @Test
+    public void testClassesInApiDoNotDependOnClassesInUtil() {
+        final JavaClasses importedClasses = new ClassFileImporter()
+            .withImportOption(ImportOption.Predefined.DO_NOT_INCLUDE_TESTS)
+            .importPackages("com.puppycrawl.tools.checkstyle.api");
+
+        final DescribedPredicate<JavaClass> suppressed = new DescribedPredicate<>("suppressed") {
+            @Override
+            public boolean apply(JavaClass input) {
+                return API_CLASSES_DEPENDENT_ON_UTILS.contains(input.getFullName());
+            }
+        };
+
+        final ArchRule testClassesInApiDoNotDependOnClassesInUtil = noClasses()
```

Comment:
ArchRule name should read like a rule, not a test: `final ArchRule ClassesInApiDoNotDependOnClassesInUtil`

Our rule is that "Classes in Api (package) Do Not Depend On Classes In Util (package)".

Reply:
Good point, changed to `noClassesInApiShouldDependOnClassesInUtil`